common/modules/buffer.jsm

```diff
@@ -731,9 +731,14 @@ var Buffer = Module("Buffer", {
         }
 
         let link = DOM("link[href][rev=canonical], \
+                        link[href][rel=shorturl], \
                         link[href][rel=shortlink]", doc)
                        .filter(sane)
                        .attr("href");
+
+        if (!link)
+            link = DOM("link[href][rel=canonical]", doc).filter(sane).attr("href");
```

Comment:
Right, that's why it would only be used if no shortened version was found. The canonical URL is then probably still much better than the one currently displayed.